quickwit-indexing/src/source/mod.rs

```diff
@@ -198,3 +198,28 @@ pub struct SourceConfig {
     pub source_type: String,
     pub params: serde_json::Value,
 }
+
+impl SourceConfig {
+    pub async fn check(&self) -> anyhow::Result<()> {
+        match &*self.source_type {
+            "file" => {
+                if let Some(path) = self.params.get("filepath").and_then(|v| v.as_str()) {
+                    if !std::path::Path::new(path).exists() {
+                        anyhow::bail!("Source `{}` does not exist", path)
+                    }
+                }
+                Ok(())
+            }
+            #[cfg(feature = "kafka")]
+            "kafka" => kafka_source::check(self.params.clone()),
+            "vec" => Ok(()),
+            unrecognized_source_type => {
```

Comment:
I'd return an error here: `Source type `{}` is not supported`. We'll move this check to `IndexConfig::validate` eventually (#757).

Reply:
My idea here was that the checks are done on a best effort basis to make the program fail fast, and are not relied upon for the further running of the application. So making this an error will require everyone who writes a new source type to extend that function immediately to be able to run their code. On the other hand, making this a warning means it can still be searched in the code, but the programmer writing the extension still have the decision when to tackle it.